libarchive/archive_read_support_format_cpio.c

```diff
@@ -633,6 +633,13 @@ header_newc(struct archive_read *a, struct cpio *cpio,
 	/* Pad name to 2 more than a multiple of 4. */
 	*name_pad = (2 - *namelength) & 3;
 
+	/* Make sure that the padded name length fits into size_t. */
+	if ((size_t)(*namelength + *name_pad) < *namelength) {
```

Comment:
Arithmetic overflow (signed or unsigned) is "undefined behavior" according to the C standard, so _cannot_ reliably be detected.

My proposed alternate form does not rely on detecting overflow.